### ISSUE TITLE
Desktop/Start Menu shortcuts use nvda_slave.exe which doesn't accept all command-line switches.

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#11681]
### Is your feature request related to a problem? Please describe.

When installing NVDA, NSIS currently points to the "nvda_slave.exe" executable for both Desktop and Start Menu shortcuts which doesn't accept all of the command-line switches that the main "nvda.exe" executable does. Could we consider changing the shortcut so that we point to "nvda.exe" instead?

**Comment by Adriani90** [comment]
cc: @leonardder, @JulienCochuyt

**Comment by Adriani90** [comment]
cc: @gerald-hartig, @michaelDCurran for triaging.

**Comment by CyrilleB79** [comment]
@elliott94 or anyone else:
The desktop or start menu shortcut use a hard-coded command line provided by the installer. The user does not pass any extra flag when executing them. Except of course if you want to define a new shortcut (or re-defining it) with different flags. Could you please clarify the use case explaining what you are want to achieve and how you are trying to do it? (e.g. which flag you are putting in the shortcut)

**Comment by nvdaes** [comment]
My shortcut in the start menu seems to use nvda, but on my desktop I have a shortcut using nvda_slave.exe.
This is an inconsistency.
@seanbudd @SaschaCowley , should this issue be triaged?
Thanks.

**Comment by CyrilleB79** [comment]
Before triaging, it would be nice to clarify the two following points:

1. We need an answer from @elliott94 regarding the use case, asked in https://github.com/nvaccess/nvda/issues/11681#issuecomment-2469753653.

2. We need to understand if the difference pointed by @nvdaes in https://github.com/nvaccess/nvda/issues/11681#issuecomment-4413553617 is on purpose or not. The desktop shortcut uses "nvda_slave.exe launchNVDA", while the startup menu shortcut only uses "nvda.exe". In the code, we can see that "nvda_slave.exe launchNVDA" launches NVDA in another process. Maybe the difference is due to the fact that the desktop shortcut is the one used by `control+alt+N` shortcut?

**Comment by nvdaes** [comment]
@CyrilleB79 wrote:

> We need an answer from @elliott94 regarding the use case, asked in Desktop/Start Menu shortcuts use nvda_slave.exe which doesn't accept all command-line switches. #11681 (comment).

I can provide a reason regarding use cases. It's difficult to answer with specific cases, since seems that the nvda_slave.exe VS nvda.exe functionality is not documented in the user guide, nor in changes.md.
Also, I remembered that Mick wrote about this in an email and I hae searched it on my inbox.
This is a quitation of part of a message submitted to the old NVDA development mailing list hosted on sourceforje:

@michaelDCurran wrote:
27 jul 2013, 6:17

> I think the only reason we use nvda_slave.exe to launch NVDA is to get 
around a feature---bug in Windows where trying to start an executable 
with a Windows shortcut key (e.g. control+alt+n) fails to run if that 
executable is already running.

End of quotation.
I have tried creating a shortcut key to nvda.exe on the desktop, with `control+alt+n`. Then I have pressed control+alt+n several times while NVDA is running, and works well here on Windows 11.
Someone on a spanish mailing list has removed his shortcut due to an error, and now I'm not sure if we should advice to create a shortcut to nvda.exe or nvda_slave.exe in the install directory (NVDAState.WritePaths.installDir).
I've suggested to get the install directory from the Python console, using `api.copyToClip` to paste the directory on the run dialog, to locate nvda.exe and create a shortcut. If this is not advisable, please let me know, but perhaps it's much better than providing an exact folder, which may vary from computers.

**Comment by nvdaes** [comment]
I think that we should use just nvda.exe in the desktop and the Start I may try to create a PR to use the shortcut available for the start menu, also for the desktop, unless the mentioned bug hasn't been fixed on Windows, though I think that the bug should be fixed.
CC: @josephsl

**Comment by seanbudd** [comment]
No, `nvda_slave.exe` is still necessary. We just need the ability to pass on arguments

**Comment by XLTechie** [comment]
> No, nvda_slave.exe is still necessary. We just need the ability to pass on arguments

@seanbudd can you provide any documentation or an STR demonstrating this?

I have personally tested, and found no obvious reason why nvda_slave.exe needs to be called here. I am wondering what test I have missed.

**Comment by michaelDCurran** [comment]
In Windows 7 at least, if you had started NVDA via the desktop shortcut, 
pressing the shortcut again would not restart NVDA as Explorer had 
tracked that it was already running.

On Windows 11 this no longer seems to be an issue. Assuming there is no 
issue on Windows 10 either, then  we can stop using nvda_slave.exe to 
start NvDA I guess.

**Comment by nvdaes** [comment]
I'll wait until the documentation requested by @XLTechie is provided, or a decision is adopted about this. Then I may consider if I can create a PR.

**Comment by seanbudd** [comment]
I don't think documentation is necessary, it seems a PR to stop using the slave for shortcuts is okay. It will be worth testing file registration (e.g. opening an addon file with NVDA)

**Comment by nvdaes** [comment]
I'm testing, and seems that the launch NVDA argument is needed to use control+alt+n while NVDA is running, but this argument is not recognized by nvda.exe, so perhaps we still need to use nvda_slave.exe in some cases, for example, if NVDA is run from source.
For now, I won't create a PR for this since results aren't always successful. Once I haven't needed the launch nvda argument.

**Comment by XLTechie** [comment]
@nvdaes In which context, exactly, does nvda_slave.exe with LaunchNVDA work, where nvda.exe with no argument fails?
Is it only when running from source?

I have tried several scenarios where just nvda.exe is used to run from shortcut, 
and have not found one (in Windows 11) that fails.

**Comment by nvdaes** [comment]
@XLTechie  wrote:

In which context, exactly, does nvda_slave.exe with LaunchNVDA work, where nvda.exe with no argument fails?
Is it only when running from source?

I tested nvda.exe with -r argument, after installing NVDA from source code. I reinstalled NVDA from source and then I found the failure regarding "launcher is not recognized".

I think that the main problem here is that it's not easy to test this in different scenarios, unless someone have a virtual machine or several computers, since of course we don't want to use the installed version of NVDA from source.

**Comment by sativarg-tuta** [comment]
Hello,
A recent question at NVDA forum has me here. I have often used my own shortcut with the target being the main NVDA.exe with no R option and NVDA restarts fine. Is the NVDA_slave.exe retained for compatibility with an older version of Windows?
Or... does the Slave.exe file deal with situations when NVDA is running with High priority and or when NVDA is frozen, unresponsive etc.
thanks
chuck
